src/vs/workbench/contrib/terminal/browser/terminalView.ts

```diff
@@ -272,6 +272,7 @@ export class TerminalViewPane extends ViewPane {
 			case TerminalCommandId.New: {
 				if (action instanceof MenuItemAction) {
 					const actions = getTerminalActionBarArgs(TerminalLocation.Panel, this._terminalProfileService.availableProfiles, this._getDefaultProfileName(), this._terminalProfileService.contributedProfiles, this._terminalService, this._dropdownMenu);
+					this._registerDisposableActions(actions.dropdownAction, actions.dropdownMenuActions);
```

Comment:
I think these function are called multiple times, so we can't register them against the TerminalView's lifecycle like this

Reply:
ok, should I add them to a disposable store and call `clear` whenever this is rerun?

Reply:
Yeah something like that, stored in a `MutableDisposable`?